### PR TITLE
Merged MatchDataProvider and DatabaseProvider into DataProvider

### LINKED ISSUE #197: Combine MatchDataProvider and DatabaseProvider
Currently, we are using React Context which pretty much acts as a "Global State" across files.

We have a MatchDataProvider, accessing the Firebase Database (Cloud Firestore) for matches. We also have a DatabaseProvider (poorly named), accessing the Firebase Storage for logos.

Please move logic from DatabaseProvider into MatchDataProvider. Update references towards the Context and layout.js, which currently uses both:

image

### PR DESCRIPTION
**Pull request by bzhang102**
Closes #197 
- Combined MatchDataProvider and DatabaseProvider contexts into single new context named DataProvider
- Renamed/repointed all occurrences to point
- Tested locally
image

### CODE REVIEW

**Review by Fredenck — CHANGES_REQUESTED** [review]
mostly looks good. Make sure you run `git merge development` so your local changes are synced with development. The filtering for schools should be live: this decreases initial load time as we only display the latest school played rather than waiting to render all of the previous schools we've played

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
   "editor.formatOnSave": true,
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "editor.codeActionsOnSave": {
-    "source.fixAll": true, // Auto-fix ESLint issues on save
-    "source.fixAll.eslint": true // Specific to ESLint
+    "source.fixAll": "explicit",
+    "source.fixAll.eslint": "explicit"
```
[the remark this answers is not in the document]
Comment (bzhang102) [reply .vscode/settings.json]: ok, merge conflict fixed, should be okay now. can you take a look to see if it's done right? want to be sure i didn't break anything lol

**Review by Fredenck — CHANGES_REQUESTED** [review]
make sure you test your changes with `npm run dev`. looks like you missed updating the import statement for match-list.

**Review by Fredenck — APPROVED** [review]
nice

**Pull request merged**